experiments/tf_trainer/tf_gru_attention/run.py

```diff
@@ -44,27 +51,40 @@ def main(argv):
   del argv  # unused
 
   embeddings_path = FLAGS.embeddings_path
+  is_binary_embedding = FLAGS.is_binary_embedding
   text_feature_name = FLAGS.text_feature_name
+  key_name = FLAGS.key_name
 
-  preprocessor = text_preprocessor.TextPreprocessor(embeddings_path)
-  nltk.download("punkt")
-  tokenize_op = preprocessor.tokenize_tensor_op(nltk.word_tokenize)
+  preprocessor = text_preprocessor.TextPreprocessor(embeddings_path, is_binary_embedding)
+  if FLAGS.preprocess_in_tf:
+    tokenize_op_init = lambda: preprocessor.tokenize_tensor_op_tf_func()
+  else:
+    nltk.download("punkt")
+    tokenize_op_init = lambda: preprocessor.tokenize_tensor_op_py_func(nltk.word_tokenize)
 
   dataset = tfrecord_input.TFRecordInput(
       train_path=FLAGS.train_path,
       validate_path=FLAGS.validate_path,
       text_feature=text_feature_name,
       labels=LABELS,
-      feature_preprocessor=tokenize_op,
+      feature_preprocessor_init=tokenize_op_init,
       batch_size=FLAGS.batch_size)
 
+  # TODO: Move embedding *into* Keras model.
```

Comment:
What does this mean?

Reply:
This comment was written by someone else.

I am assuming that the goal here would be to have the embeddings directly in the model function (keras and tf model.py) and therefore get rid of this function add_embedding_to_model.